scripts/qemu_launcher.sh

```diff
@@ -0,0 +1,159 @@
+# It wraps up the kvm command and provides simple options to lanuch dev and prod images
+# without loosing the flexibility to adjust parameters like memory, cpu, etc.
+#
+# To see the usage, run:
+# ./qemu_launcher.sh -h
+#
+# e.g.
+# To start dev image with static IP (asumming the current folder is the build)
+# ./qemu_launcher.sh -d -s
+#
+#
+
+command -v kvm > /dev/null
+if [ $? != 0 ]
+then
+    echo "Warning:"
+    echo "kvm is not installed yet, please install it with command below and rerun the script"
+    echo "sudo apt-get install qemu-kvm libvirt-clients libvirt-daemon-system "
+    exit 1
+fi
+
+set -e
+
+# Print usage information of the script
+usage() {
+    echo "usage: $0 [options]"
+    echo -e "[options] is any of the following:"
+    echo -e "  -f | --folder           \tSpecify a source folder of the QEMU image"
+    echo -e "                          \tDefaults to tmp/deploy/images/qemux86-64"
+    echo -e "  -i | --hda              \tSpecify rootfs file. Defaults to core-image-pelux-minimal-qemux86-64.ext4"
+    echo -e "  -p | --ssh_port         \tSSH port to connet to QEMU. Defaults to 1234"
+    echo -e "  -n | --smp              \tNumber of cpu cores. Defaults to 4"
+    echo -e "  -k | --kernel_path      \tBy default, kernel_path is the same as folder, but can be overwritten by using this option"
+    echo -e "  -c | --cmdline          \tSpecify kernel command line. This would allow power users to change vga or root, etc."
+    echo -e "                          \tDefaults to \"vga=0 uvesafb.mode_option=1280×720-32 root=/dev/hda console=ttyS0 rw oprofile.timer=1\""
+    echo -e "  -m | --mem              \tMaximum amount of guest memory (Size is in megabytes)"
+    echo -e "                          \tDefaults to 512"
+    echo -e "  -t | --gdb_port         \tThe port is forwarded so that gdbserver can listen on it"
+    echo -e "                          \tThen gdb can be run from develop machine to connect to the port for remote debugging"
+    echo -e "                          \tDefaults to 3333"
+    echo -e "  -d | --dev              \tRun QEMU development image instead of normal image"
+    echo -e "                          \tDefaults to core-image-pelux-minimal-dev-qemux86-64.ext4"
+    echo -e "  -s | --static           \tRun QEMU development image with static IP (192.168.7.2)"
+    echo -e "  -l | --list             \tList all available images from the default folder or folder user specified"
+    echo -e "  -h | --help             \tDisplay this help"
+    exit 0
+}
+
+list_images() {
+    echo "Available images:"
+    ls $folder/*.ext4 | xargs -n 1 basename
+    exit 0
+}
+
+setup_folder() {
+    folder=$1
+    if [ -z ${kernel_path+x} ]
+    then
+        kernel_path=$folder
+    fi
+}
+
+setup_dev() {
+    DEV=true
+    if [ $hda == "core-image-pelux-minimal-qemux86-64.ext4"  ]
+    then
+        hda="core-image-pelux-minimal-dev-qemux86-64.ext4"
+    fi
+}
```

Comment:
Sorry but what does this code do? If hda is set, we set it again to the same value?